library/ecp.c

```diff
@@ -410,6 +410,12 @@ static const mbedtls_ecp_curve_info ecp_supported_curves[] =
 #endif
 #if defined(MBEDTLS_ECP_DP_SECP192K1_ENABLED)
     { MBEDTLS_ECP_DP_SECP192K1,    18,     192,    "secp192k1"         },
+#endif
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
```

Comment:
I just noticed, that we have a different problem here. These curves are automatically offered by TLS in the handshake, but the Mbed TLS x25519 implementation is not integrated completely yet, and the handshake will fail if this curve is selected by the peer. I think that for now we need to condition this on the presence of EVEREST.

Reply:
I think we should leave it, but fix `mbedtls_ecdh_can_do` and `mbedtls_ecdsa_can_do`.

Reply:
If we do that, we need to change the mechanism used by the SSL module uses to decide about which curves to offer to the peers. Currently it takes this list, unless the user overrides it by manually editing the SSL context. To resolve this problem we would need to change/override this dynamically, based on what `mbedtls_ecdh_can_do()` says. We might want to do that eventually, but it feels out of scope for this PR and adding EVEREST as a compile time guard would be a one line change and we could refactor it when we complete support for our own Montgomery curve implementation.

Reply:
Makes sense. So the best solution (for now) is to guard the x25519 entry in this list with `MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED` and to remove the entry for x448?

Reply:
Yes, that would be my preferred option. @gilles-peskine-arm are you happy with this solution for now?

Reply:
I'm not _happy_ about it, but it does seem like the best temporary solution, so ok.

Reply:
Done! 202d06b74071e1d9f63ef4b49ad669f42b3f3e26